### PR TITLE
Add content negotiation

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Seems to be suffering the same problem as #163.

File: composer.json [hunk 1]
```diff
@@ -35,6 +36,7 @@
         "symfony/event-dispatcher": "dev-master",
         "symfony/monolog-bridge": "^3.2",
         "symfony/psr-http-message-bridge": "^0.2.0",
+        "willdurand/negotiation": "^2.3",
```
Comment (giorgiosironi) [inline composer.json]: isn't this dependency brought over from `elife/content-negotiator`?

Comment (thewilkybarkid) [reply composer.json]: Yes, but it's touched in https://github.com/elifesciences/search/blob/7c5b5e1f10bca4b0e91740cc17199ad735c8dced/src/Search/Api/SearchController.php#L17

**Comment by giorgiosironi** [comment]
Same problem:
```
[2017-11-14 09:13:57,936][DEBUG][action.search            ] [search--ci] All shards failed for phase: [query]
RemoteTransportException[[search--ci][10.0.2.189:9300][indices:data/read/search[phase/query]]]; nested: IllegalIndexShardStateException[Cur
rentState[RECOVERING] operations only allowed when shard state is one of [POST_RECOVERY, STARTED, RELOCATED]];
Caused by: [elife_test][[elife_test][1]] IllegalIndexShardStateException[CurrentState[RECOVERING] operations only allowed when shard state 
is one of [POST_RECOVERY, STARTED, RELOCATED]]
        at org.elasticsearch.index.shard.IndexShard.readAllowed(IndexShard.java:1034)
        at org.elasticsearch.index.shard.IndexShard.acquireSearcher(IndexShard.java:848)
        at org.elasticsearch.search.SearchService.createContext(SearchService.java:656)
        at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:633)
        at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:377)
        at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:368)
        at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:365)
        at org.elasticsearch.transport.TransportRequestHandler.messageReceived(TransportRequestHandler.java:33)
        at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:77)
        at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:376)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by giorgiosironi** [comment]
May be a race condition where we don't wait enough when creating the index. I think it is recreated for each test to start from an empty dataset.

**Pull request merged**